### ISSUE TITLE
Support creating NuGet packages using both .csproj info and NuGetPackSettings info

### ISSUE THREAD

**Issue by pvwichen** [cake-build/cake#1037]
### What You Are Seeing?
Using the `NuGetPack(FilePath filePath, NuGetPackSettings settings)` command where filePath is pointed at a `.csproj` file, you can create a NuGet package with the dependencies defined based on the .csproj file references. However, the NuGetPackSettings (i.e. the `settings` parameter) are only partially used. A .nuspec file is not created and the settings are only used to determine command line parameters. Settings like the `Files` property are ignored.

Using the `NuGetPack(NuGetPackSettings settings)` command, you can create a NuGet package that uses your `NuGetPackSettings` definition (from the `settings` parameter). This allows you to specify additional files to include (e.g. symbols). However, the .csproj references are not used and must be defined in the `NuGetPackSettings`.

We use Proget which allows a regular nuget package to contain both the assemblies and symbols (i.e. you do not need a separate nuget package for symbols).

Given the current behavior above, we can create a NuGet package that has dependencies automatically defined or a NuGet package that contains symbols, but not a package that has both.

### What is Expected?
The `NuGetPack` method should allow the user to specify if the `NuGetPackSettings` should be used to generate `.nuspec` file in the same folder as the project file (i.e. if the filePath points to a project file, a .nuspec file will be automatically created in the project file's folder). The `nuget pack` command will still be passed the .csproj file as its file parameter, but the NuGet tool will then detect that the .nuspec file exists and use both .csproj/.nuspec information (refer to https://docs.nuget.org/create/creating-and-publishing-a-package in the "From a project" section).

It appears that the Cake repo already has the majority of the required code. I would propose that a new setting is added to the `NuGetPackSettings` class (e.g. `bool CreateNuspecFileForProject`).
If true and the filePath is pointed to a .csproj, the `NuGetPackSettings` would be used to create/update the .nuspec file in the project's folder prior to execute the `Run` method (in the `NuGetPacker` class).

### What version of Cake are you using?
0.13.0

### Are you running on a 32 or 64 bit system?
64-bit

### What environment are you running on?  Windows? Linux? Mac?
Linux, Windows

### Are you running on a CI Server?  If so, which one?
Jenkins, Teamcity

**Comment by pvwichen** [comment]
An alternative approach might be to expose a `NuGetSpec(FilePath nuspecFilePath, NuGetPackSettings settings)` script alias that can be called prior to `NuGetPack`. The alias would write the appropriate portions of the NuGetPackSettings to a .nuspec file (based on the specified file path).

**Comment by gep13** [comment]
@devlead what are your thoughts on this one?

**Comment by devlead** [comment]
Personally think the alias does too much as is, would prefer a new alias for this. As nuget doesn't support supplying both a spec and csproj.

**Comment by gep13** [comment]
@devlead yes, I think that makes sense to me as well.

**Comment by pvwichen** [comment]
@devlead and @gep13 : When you mention adding a new alias, is your thought to expose a `NuGetSpec(FilePath nuspecFilePath, NuGetPackSettings settings)` command or do you have a different idea in mind?
If helpful, I am willing to add the new alias based on your direction.

**Comment by devlead** [comment]
@pvwichen yes but likely NuGetSpecSettings as it is it's own nuget command.

**Comment by pvwichen** [comment]
OK - wasn't sure about that originally since they would share a significant amount of settings, but a new settings class would be better long term for maintainability. 

Based on the scenario I would like to support (i.e. save a nuspec file to disk from the script and then execute `NuGetPack` with the .csproj file path), maybe it would be responsible to add an extension method that creates a `NuGetPackSettings` instance from a `NuGetSpecSettings` instance, especially since they would share a significant number of properties.

For now, can just start with the new alias / settings. Let me know if I can help with this.

**Comment by luisgoncalves** [comment]
Related to #872

**Comment by IanMercer** [comment]
We create a .nuspec file next to each csproj file in source code. We have an alias that finds all the nuspec files but translates each file path to the matching csproj file. We run nugetpack using the csproj file. This works well and allows developers to configure which projects get packaged up as Nugets. The nuspec file remains under source control not some ephemeral file on the build server. By using $id$, $version$ etc. the nuspec file is almost identical for every project.

Here's the alias:

        [CakeMethodAlias]
        public static FilePathCollection GetProjectFilesWithNuspecs(this ICakeContext context, string path = "**/*.nuspec")
        {
            var nuspecs = context.GetNuspecFiles(path);
            var csprojs = nuspecs.Select(nu => nu.ChangeExtension(".csproj"))
                .Where(filePath => System.IO.File.Exists(filePath.FullPath))
                .ToList();
            return new FilePathCollection(csprojs, PathComparer.Default);
        }

**Comment by devlead** [comment]
@pvwichen checkin status of this issue, are you in a position to do a PR for this?

**Comment by pvwichen** [comment]
Yes... I should be able to complete that. Based on the prior discussion, my plan would be basically to:

- Create an alias: `NuGetSpec(FilePath nuspecFilePath, NuGetSpecSettings settings)`
- Create a new settings class `NuGetSpecSettings`.
- Create method to map from `NuGetPackSettings` to `NuGetSpecSettings`.

Does that sound about right?

**Comment by devlead** [comment]
@pvwichen yes that sounds reasonable.

**Comment by aaronasmith** [comment]
Did anything ever come of this?  We would love to have this functionality implemented as well.

**Comment by pvwichen** [comment]
No... I am not in a position to do this anytime soon so it is up for grabs.
